### PR TITLE
Improve HTTP detection heuristic

### PR DESCRIPTION
**Pull request by fatanugraha**
#### Description

Improve the HTTP detection heuristic introduced in this PR: https://github.com/mitmproxy/mitmproxy/pull/7087

In my mitmproxy `transparent` mode setup, [testcontainer's](https://testcontainers.com/getting-started/) ryuk breaks because it sends the string below in raw TCP
```
label=org.testcontainers.version=0.30.0&label=org.testcontainers.sessionId=eb48d24ade2174c56e8951a306241ad68eccdd8fcb7686b62ed03e638ec73870&label=org.testcontainers=true&label=org.testcontainers.lang=go\n
```

Upon checking the proxy log, I found out that mitmproxy thinks that it's an HTTP protocol and try to parse it as such, causing the connection to appear as "stuck"

#### Checklist

 - [x] I have updated tests where applicable.
 - [x] I have added an entry to the CHANGELOG.

### CODE REVIEW

File: mitmproxy/addons/next_layer.py [hunk 1]
```diff
@@ -184,6 +184,8 @@ def s(*layers):
             len(data_client) < 3
             # HTTP would require whitespace before the first newline
             # if we have neither whitespace nor a newline, it's also unlikely to be HTTP.
+            or (data_client.find(b" ") == -1)
+            or (data_client.find(b"\n") == -1)
```
Comment (mhils) [inline mitmproxy/addons/next_layer.py:188]: We can have incomplete request lines in the first packet, so I would rather go for something like this:

```suggestion
            or ("\n" in data_client and " " not in data_client)
```

What do you think?

File: CHANGELOG.md [hunk 2]
```diff
@@ -23,7 +25,7 @@
   ([#6720](https://github.com/mitmproxy/mitmproxy/pull/6720), @NicolaiSoeborg)
 - Fix slow tnetstring parsing in case of very large tnetstring.
   ([#7121](https://github.com/mitmproxy/mitmproxy/pull/7121), @mik1904)
-- Add `getaddrinfo`-based fallback for DNS resolution if we are unable to 
+- Add `getaddrinfo`-based fallback for DNS resolution if we are unable to
```
Comment (mhils) [inline CHANGELOG.md:28]: Please don't add unrelated changes here.

**Review by mhils — COMMENTED** [review]
Thanks!

File: mitmproxy/addons/next_layer.py (hunk 1 above)
Comment (fatanugraha) [reply mitmproxy/addons/next_layer.py]: That makes sense, thanks!

**Review by mhils — APPROVED** [review]
Thanks! 🍰
